app/services/search/location_builder.rb

```diff
@@ -21,44 +20,37 @@ def initialize(location, radius, location_category, buffer_radius)
                          end
 
     if NATIONWIDE_LOCATIONS.include?(@location&.downcase)
-      initialize_nationwide_search
+      @location = nil
     elsif location_category_search?
-      initialize_location_polygon
+      initialize_polygon_boundaries
     elsif @location.present?
       @location_filter = build_location_filter(@location, @radius)
     end
   end
 
   def location_category_search?
-    (@location_category && LocationCategory.include?(@location_category)) ||
-      (@location && LocationCategory.include?(@location))
+    (location_category && LocationCategory.include?(location_category)) ||
+      (location && LocationCategory.include?(location))
   end
 
   private
 
-  def initialize_location_polygon
-    @location_polygon = LocationPolygon.with_name(@location_category)
-    if @location_polygon.present?
-      @search_polygon_boundary = if @buffer_radius.present?
-                                   [@location_polygon.buffers[@buffer_radius]]
-                                 else
-                                   [@location_polygon.boundary]
-                                 end
-    end
+  def initialize_polygon_boundaries
+    locations = [LocationPolygon.with_name(location_category)]
 
-    return unless location_polygon.nil? && (DOWNCASE_REGIONS + DOWNCASE_COUNTIES).include?(@location_category.downcase)
+    if locations.none? && DOWNCASE_COMPOSITE_LOCATIONS.key?(location_category.downcase)
+      locations = DOWNCASE_COMPOSITE_LOCATIONS[location_category.downcase].map do |component_location_name|
+        LocationPolygon.find_by(name: component_location_name.downcase)
+      end
+    end
 
-    # If a location category that we expect to have a polygon actually does not,
-    # append the location category to the text search as a fallback.
-    # This applies only to regions and counties: large areas for which there is
-    # very little value in using a point coordinate, and for which there is a
-    # low chance of ambiguity (unlike Clapham borough vs Clapham village in Bedfordshire)
-    Rollbar.log(
-      :error,
-      "A location category search was performed as a text search as no LocationPolygon could
-      be found with the name '#{@location_category}'.",
-    )
-    @missing_polygon = true
+    @polygon_boundaries = []
+    locations.each do |location|
+      polygons = buffer_radius.present? ? location.buffers[buffer_radius] : location.polygons["polygons"]
+      polygons.each do |polygon|
+        @polygon_boundaries.push(polygon)
```

Comment:
`polygon_boundaries`

Reply:
Warum? I'm trying to change the attributes value.

Reply:
Maybe I should learn to **read** before I open my mouth 😂